### PR TITLE
[Substrate Connectors] [Beta] Document breaking changes for beta namespace

### PR DESCRIPTION
**Pull request by emzho**
For this documentation change, we are updating API references for Search APIs to:
1. Make sure namespace references are updated from microsoft.graph --> microsoft.graph.externalConnectors
2. Remove @odata.type references from code examples
2. Make sure property name is updated from authorizedApps --> authorizedAppIds
3. Update "property" enum to be camel-cased
4. Update error object to be graph.publicError object
5. Update table of contents for Search APIs

### CODE REVIEW

File: api-reference/beta/resources/externalitem.md [hunk 1]
```diff
@@ -44,18 +45,27 @@ The following is a JSON representation of the resource.
...
+      "type": "everyone",
+      "value": "67a141d8-cf4e-4528-ba07-bed21bfacd2d",
+      "accessType": "grant",
+      "identitySource": "azureActiveDirectory"
+    }
+  ],
   "id": "String (identifier)",
-  "properties": "Object"
+  "properties": "Object",
+  "content": {
```
Comment (millicentachieng) [inline api-reference/beta/resources/externalitem.md:65]: Change this back to `"content": {"@odata.type": "microsoft.graph.externalItemContent"}`

File: api-reference/beta/api/externalconnectors-connectionoperation-get.md [hunk 2]
```diff
@@ -45,7 +46,7 @@ Do not supply a request body for this method.
 
 ## Response
 
-If successful, this method returns a `200 OK` response code and the requested [connectionOperation](../resources/connectionoperation.md) object in the response body.
+If successful, this method returns a `200 OK` response code and the requested [connectionOperation](../resources/externalconnectors-connectionoperation.md) object in the response body.
```
Comment (jasonjoh) [inline api-reference/beta/api/externalconnectors-connectionoperation-get.md:49]: Renaming files will change their URL. Any links to the old URL will 404 unless we set redirects.

File: api-reference/beta/api/externalconnectors-externalconnection-list.md [hunk 3]
```diff
@@ -138,4 +141,5 @@ Content-type: application/json
   "tocPath": ""
 }-->
 
-
+> [!NOTE]
```
Comment (jasonjoh) [inline api-reference/beta/api/externalconnectors-externalconnection-list.md:144]: This needs more context. Just looking at this topic, this is the only mention of the `authorizedApps` property, so it seems confusing to bring it up at all. A note about the property being deprecate may make more sense in the resource topic for `externalConnection`.

File: api-reference/beta/api/externalconnectors-externalitem-delete.md [hunk 4]
```diff
@@ -37,8 +38,8 @@ DELETE /external/connections/{connection-id}/items/{item-id}
 
 | Parameter     | Type   | Description                                         |
 |:--------------|:-------|:----------------------------------------------------|
-| connection-id | string | The `id` property of the containing [externalConnection](../resources/externalconnection.md) |
-| item-id       | string | The developer-provided `id` property of the [externalItem](../resources/externalitem.md). |
+| connection-id | String | The `id` property of the containing [externalConnection](../resources/externalconnectors-externalconnection.md) |
```
Comment (jasonjoh) [inline api-reference/beta/api/externalconnectors-externalitem-delete.md:41]: lowercase `string`

File: api-reference/beta/api/externalconnectors-externalconnection-put-items.md [hunk 5]
```diff
@@ -39,8 +40,8 @@ PUT /external/connections/{connection-id}/items/{item-id}
 
 | Parameter     | Type   | Description                                         |
 |:--------------|:-------|:----------------------------------------------------|
-| connection-id | string | The `id` property of the containing [externalConnection](../resources/externalconnection.md) |
-| item-id       | string | The developer-provided `id` property of the [externalItem](../resources/externalitem.md). If no item already exists with this `id`, a new item is created. If an item already exists with this `id`, it is overwritten by the object sent in the body. |
+| connection-id | String | The `id` property of the containing [externalConnection](../resources/externalconnectors-externalconnection.md) |
```
Comment (jasonjoh) [inline api-reference/beta/api/externalconnectors-externalconnection-put-items.md:43]: lowercase `string`

File: api-reference/beta/api/externalconnectors-externalitem-update.md [hunk 6]
```diff
@@ -37,8 +38,8 @@ PATCH /external/connections/{connection-id}/items/{item-id}
 
 | Parameter     | Type   | Description                                         |
 |:--------------|:-------|:----------------------------------------------------|
-| connection-id | string | The `id` property of the containing [externalConnection](../resources/externalconnection.md) |
-| item-id       | string | The developer-provided `id` property of the [externalItem](../resources/externalitem.md). |
+| connection-id | String | The `id` property of the containing [externalConnection](../resources/externalconnectors-externalconnection.md) |
```
Comment (jasonjoh) [inline api-reference/beta/api/externalconnectors-externalitem-update.md:41]: lowercase `string`

File: api-reference/beta/resources/externalconnectors-configuration.md [hunk 7]
```diff
@@ -3,23 +3,24 @@ title: "configuration resource type"
...
 
-Specifies additional application IDs that are allowed to manage the externalConnection and to index content in a [externalConnection](../resources/externalconnection.md).
+Specifies additional application IDs that are allowed to manage the externalConnection and to index content in a [externalConnection](../resources/externalconnectors-externalconnection.md).
 
 ## Properties
 
 | Property       | Type              | Description |
 |:---------------|:------------------|:------------|
-| authorizedApps | String collection | A collection of application IDs for registered Azure Active Directory apps that are allowed to manage the externalConnection and to index content in the externalConnection. |
+| authorizedAppIds | String collection | A collection of application IDs for registered Azure Active Directory apps that are allowed to manage the externalConnection and to index content in the externalConnection. |
```
Comment (jasonjoh) [inline api-reference/beta/resources/externalconnectors-configuration.md:23]: Here would probably be a good place to put the note about renaming the property.

File: api-reference/beta/api/externalconnectors-connectionoperation-get.md (hunk 2 above)
Comment (emzho) [reply api-reference/beta/api/externalconnectors-connectionoperation-get.md:49]: Yep! I think I have resolved all the file renames and hyperlinks for this. API doctor / other validation flagged broken hyperlinks for us, which was helpful.

**Review by jasonjoh — CHANGES_REQUESTED** [review]

File: api-reference/beta/api/externalconnectors-connectionoperation-get.md (hunk 2 above)
Comment (jasonjoh) [reply api-reference/beta/api/externalconnectors-connectionoperation-get.md:49]: I added redirect entries to the .openpublishing.redirect.json so if there are any links out there to the old versions they won't be broken. Can you give the changes to that file a quick glance to see if I missed any?

File: api-reference/beta/api/externalconnectors-connectionoperation-get.md [hunk 8]
```diff
@@ -3,17 +3,18 @@ title: "Get connectionOperation"
 description: "Retrieve the properties of a connectionOperation."
 localization_priority: Normal
 author: "snlraju-msft"
+ms.author: mecampos
```
[the remark this answers is not in the document]
Comment (emzho) [reply api-reference/beta/api/externalconnectors-connectionoperation-get.md:6]: Ok, will do.

File: api-reference/beta/api/externalconnectors-externalconnection-list.md (hunk 3 above)
Comment (emzho) [reply api-reference/beta/api/externalconnectors-externalconnection-list.md:144]: @snlraju-msft can you help with this?

**Comment by jasonjoh** [comment]
Remaining warnings about .openpublishing.redirection.json are caused by values that were already in the file prior to my edits. Opened https://github.com/microsoftgraph/microsoft-graph-docs/issues/13528 to track cleanup of that file as a separate work item.

File: api-reference/beta/resources/externalconnectors-configuration.md (hunk 7 above)
Comment (emzho) [reply api-reference/beta/resources/externalconnectors-configuration.md:22]: @snlraju-msft -- leaving this for you 👍

File: api-reference/beta/resources/externalconnectors-externalitem.md [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+    {
+      "type": "everyone",
+      "value": "67a141d8-cf4e-4528-ba07-bed21bfacd2d",
+      "accessType": "grant",
+      "identitySource": "azureActiveDirectory"
+    }
+  ],
+  "id": "String (identifier)",
+  "properties": "Object",
+  "content": {
```
Comment (millicentachieng) [inline api-reference/beta/resources/externalconnectors-externalitem.md:64]: Remove `value` and `type` properties. Having `"@odata.type": "microsoft.graph.externalConnectors.externalItemContent"` already indicates these two properties are expected.

File: api-reference/beta/api/externalconnectors-externalconnection-list.md (hunk 3 above)
Comment (snlraju-msft) [reply api-reference/beta/api/externalconnectors-externalconnection-list.md:143]: Resolving this as this is covered in the breaking changes blog.

File: api-reference/beta/resources/externalconnectors-externalitem.md (hunk 9 above)
Comment (jasonjoh) [reply api-reference/beta/resources/externalconnectors-externalitem.md:64]: Taking care of it.

File: api-reference/beta/api/externalconnectors-externalconnection-list.md (hunk 3 above)
Comment (jasonjoh) [reply api-reference/beta/api/externalconnectors-externalconnection-list.md:143]: Removed per our convo

File: api-reference/beta/resources/externalconnectors-configuration.md (hunk 7 above)
Comment (jasonjoh) [reply api-reference/beta/resources/externalconnectors-configuration.md:22]: Added note.

**Review by jasonjoh — APPROVED** [review]

**Pull request merged**